src/main/scala/example/BookBot.scala

```diff
@@ -0,0 +1,44 @@
+package example
+
+import scalaj.http._
+import io.circe._, io.circe.parser._
+import example.Book
+
+case class TransformError(message: String)
+case class HttpError(message: String)
```

Comment:
I like this pattern of having specific `case class` types for different errors, makes it very readable and adds type safety 🥇 . Only thing I'd add is to potentially create a `models` or a `dto` (data transfer object) directory to house all our case classes, and in that having one directory for errors and one for other models.

Reply:
So should i have a file for each case class?

Reply:
Potentially, let the single responsibility principle be your guide.